src/components/Sidebar/Sidebar.jsx

```diff
@@ -0,0 +1,240 @@
+  sortingCriteriaName[0] = sortingCriteriaName[0] + ":";
+  return sortingCriteriaName.join(' ');
+}
+
+export const Sidebar = () => {
+
+    const [categories, setCategories] = useState();
+    const [features, setFeatures] = useState();
+    const {state, dispatch} = useProductsAndFiltersData()
+
+
+    useEffect(() => {
+      console.log(state);
+      (async() => {
+        const recievedCategories = await axios.get(`${BACKEND_URL}/categories`);
+        setCategories(recievedCategories.data);
+        
+        const recievedFeatures = await axios.get(`${BACKEND_URL}/features`);
+        setFeatures(recievedFeatures.data);
+      })();
+    },[])
+
+    const maximumProductPrice = Math.ceil(getMaximumProductPrice(state.allProducts));
+    const minimumProductPrice = Math.floor(getMinimumProductPrice(state.allProducts));
+    const averageProductPrice = Math.floor(minimumProductPrice + (maximumProductPrice - minimumProductPrice) / 2.0);
+    const priceSliderRef = useRef();
+
+    console.log(minimumProductPrice, maximumProductPrice)
+
+    
+    return (
+        <aside className="sidebar-container fixed-position">
+        <div className="filters-heading-container row-container">
+          <h3>Clear Filters</h3>
+        </div>
+
+        <h2 className="sidebar-heading">Price</h2>
+        <div className="price-slider">
+          <div className="price-range-values-container width-100-percent">
+            <h3>{minimumProductPrice}</h3>
+            <h3>{averageProductPrice}</h3>
+            <h3>{maximumProductPrice}</h3>
+          </div>
+          <input
+            type="range"
+            min={minimumProductPrice}
+            max={maximumProductPrice}
+            defaultValue={maximumProductPrice}
+            id="price-slider"
+            className="slider width-100-percent"
+            ref={priceSliderRef}
+            onChange={
+              (event) => {
+                dispatch({
+                  ...state,
+                  actionType: ACTION_TYPE.PRICE_FILTER,
+                  selectedPrice: event.target.value
+                })
+              }
+            }
+          />
+        </div>
+
+        <p>Displying products upto <strong>{`₹ ${priceSliderRef.current && priceSliderRef.current.value || maximumProductPrice}`}</strong></p>
+        <p>Found <strong>{state.displayedProducts.length} products</strong> satisfying current criterias.</p>
+        
+        <h2 className="sidebar-heading">Category</h2>
+        
+        <div className="categories-container">
+        {categories && categories.map(({_id,category}) => 
+              <div key={_id}>
+              <input
+                type="checkbox"
+                id={category}
+                // name="category"
+                // value="Category"
+                onChange= {
+                  (event) => {
+                    dispatch(getUpdatedStateBasedOnCategory(state, event.target.id, event.target.checked));
```

Comment:
Again, I am not too sure if passing a function to dispatch is the norm.

Reply:
The function is not being passed to dispatch, it is the return value of that function which is being passed to it